flocker/control/httpapi.py

```diff
@@ -53,6 +68,116 @@ def version(self):
         """
         return {u"flocker":  __version__}
 
+    @app.route("/datasets", methods=['POST'])
+    @user_documentation(
+        """
+        Create a new dataset.
+        """,
+        examples=[
+            u"create dataset",
+            u"create dataset with dataset_id",
+            u"create dataset with duplicate dataset_id",
+            u"create dataset with maximum_size",
+            u"create dataset with metadata",
+        ]
+    )
+    @structured(
+        inputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        schema_store=SCHEMAS
+    )
+    def create_dataset(self, primary, dataset_id=None, maximum_size=None,
+                       metadata=None):
+        """
```

Comment:
I would suggest that the bulk of the documentation be not here, but rather in the `user_documentation` decorator where it's more useful.